src/app/dao/tests/JoinerDAO.spec.ts

```diff
@@ -1,17 +1,438 @@
+    it('should forbid creating a joiner on behalf of another user', async() => {
+        // Given two users, including a malicious one, and an existing part
+        const regularUser: FireAuth.User = await createConnectedGoogleUser('foo@bar.com', 'regular');
+        const regularMinimalUser: MinimalUser = { id: regularUser.uid, name: 'regular' };
+        const partId: string = await partDAO.create(PartMocks.INITIAL);
+        await signOut();
+        await createConnectedGoogleUser('bar@bar.com', 'malicious');
+        // When the malicious user creates the corresponding joiner on behalf of the regular user
+        const result: Promise<void> = joinerDAO.set(partId, { ...JoinerMocks.INITIAL, creator: regularMinimalUser });
+        // Then it should fail
+        await expectFirebasePermissionDenied(result);
+    });
+    it('should forbid non-verified user to create a joiner', async() => {
+        // Given an existing part.
+        // (Note that the non-verified user in practice can never have created the corresponding part,
+        // but this is an extra check just in case)
+        await createConnectedGoogleUser('foo@bar.com', 'creator');
+        const partId: string = await partDAO.create(PartMocks.INITIAL);
+        await signOut();
+
+        // and a non-verified user
+        const user: FireAuth.User = await createUnverifiedUser('bar@bar.com', 'user');
+        const nonVerifiedUser: MinimalUser = { id: user.uid, name: 'user' };
+
+        // When the user tries to create the joiner
+        const result: Promise<void> = joinerDAO.set(partId, { ...JoinerMocks.INITIAL, nonVerifiedUser });
+        // Then it should fail
+        await expectFirebasePermissionDenied(result);
+    });
+    it('should allow creator to change other fields than candidates and partStatus to STARTED', async() => {
+        // Given a user that created a part and joiner
+        const user: FireAuth.User = await createConnectedGoogleUser('foo@bar.com', 'creator');
+        const creator: MinimalUser = { id: user.uid, name: 'creator' };
+        const partId: string = await partDAO.create(PartMocks.INITIAL);
+        await joinerDAO.set(partId, { ...JoinerMocks.INITIAL, creator });
+
+        const updates: Partial<Joiner>[] = [
+            { chosenPlayer: 'candidate' },
+            { partStatus: PartStatus.CONFIG_PROPOSED.value },
+            { firstPlayer: FirstPlayer.CHOSEN_PLAYER.value },
+            { partType: PartType.BLITZ.value },
+            { maximalMoveDuration: 73 },
+            { totalPartDuration: 1001001 },
+        ];
+        for (const update of updates) {
+            // When modifying a field
+            const result: Promise<void> = joinerDAO.update(partId, update);
+            // Then it should succeed
+            await expectAsync(result).toBeResolvedTo();
+        }
+    });
+    it('should forbid the creator to change the part status to STARTED', async() => {
+        // Given a user that created a part and joiner
```

Comment:
an user!

Reply:
et la fonction qui l'authorise a peut être sa place juste ici en dessous ?

Reply:
a user! user commence par une consonne: y ! (oui oui pour du vrai !)
Sinon si par "la fonction qui l'autorise" tu veux dire le test qui autorise les chosenOpponent à changer le statut, il est sous le bloc "chosenOpponent" donc pas trop déplaçable

Reply:
https://translate.google.com/?source=osdd&sl=auto&tl=en&text=un%20utilisateur&op=translate

Google translation déssacorde avec toi !

Reply:
mais les grammairiens de wordreference sont d'accord !
https://forum.wordreference.com/threads/a-an-user.480322/
De même que les nerd langagiers de stackexchange !
https://english.stackexchange.com/questions/105116/is-it-a-user-or-an-user

Je fais plus confiance à des humains qu'une IA :p